dlp/api/DlpTest/DlpTest.cs

```diff
@@ -0,0 +1,144 @@
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value), checkPhone: true, checkEmail: true, checkCc: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-i", 
+                "PHONE_NUMBER,EMAIL_ADDRESS,CREDIT_CARD_NUMBER"), checkPhone: true, checkEmail: true, checkCc: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-i", "PHONE_NUMBER,EMAIL_ADDRESS"), 
+                checkPhone: true, checkEmail: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-i", "PHONE_NUMBER,CREDIT_CARD_NUMBER"), 
+                checkPhone: true, checkCc: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-i", "EMAIL_ADDRESS,CREDIT_CARD_NUMBER"), 
+                checkEmail: true, checkCc: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-i", "PHONE_NUMBER"), checkPhone: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-i", "EMAIL_ADDRESS"), checkEmail: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-i", "CREDIT_CARD_NUMBER"), checkCc: true);
+        }
+
+        /// <summary>
+        /// Dlp evaluates this content string as having 
+        /// PHONE_NUMBER - VERY_LIKELY
+        /// EMAIL_ADDRESS - LIKELY
+        /// CREDIT_CARD_NUMBER - VERY_LIKELY
+        /// </summary>
+        [Theory]
+        [InlineData("inspectString", inspectStringValue)]
+        [InlineData("inspectFile", inspectFilePath)]
+        public void TestInspectStringFilterLikelihood(string verb, string value)
+        {
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-l", "4"), checkPhone: true, checkEmail: true, 
+                checkCc: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-l", "5"), checkPhone: true, checkCc: true);
+        }
+
+        /// <summary>
+        /// Dlp finds the matches in the order phone -> email -> credit card
+        /// </summary>
+        [Theory]
+        [InlineData("inspectString", inspectStringValue)]
+        [InlineData("inspectFile", inspectFilePath)]
+        public void TestInspectStringFilterMaxResults(string verb, string value)
+        {
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-m", "0"), checkPhone: true, checkEmail: true, 
+                checkCc: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-m", "3"), checkPhone: true, checkEmail: true, 
+                checkCc: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-m", "2"), checkPhone: true, checkEmail: true);
+            AssertPhoneEmailCC(_dlp.Run(verb, _projectId, value, "-m", "1"), checkPhone: true);
+        }
+
+        [Fact]
+        public void TestTemplates()
+        {
+            // Creation
+            ConsoleOutput output = _dlp.Run("createInspectTemplate", _projectId, "testDisplayName", "test description");
+            Assert.Contains("name: ", output.Stdout);
```

Comment:
Does the name for the created template not always match the input name?

Reply:
When creating a template, you pass a "display name", and the actual name is generated by the BE.
Honestly, I don't like these tests. We are testing CLI layer instead of the actual API. However I just followed established pattern. We should have a separate discussion on whether we need CLI layer at all, and regardless, how do we test the sample code. Both are outside of this PR.

Reply:
FWIW - the Node samples test the CLI to verify that the CLI and API don't diverge over time. If you want to drop the CLI itself, that's certainly an option (though it may make it harder for our team's tools to interact with your samples) - but if we're keeping it for now, I'm personally in favor of keeping tests as "end-to-end" as possible.